### PR TITLE
Add strict options

### PR DESCRIPTION
**Pull request by PayscaleNateW**
- Created a "strict-addons" folder with some additional configs you can extend
- Created vanilla js strict addon, for pure vanilla js rules
- Pulled in vanilla rules from https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/best-practices.js and https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/errors.js
- Intend to add strict-addons for ES6 (might include imports in this), React, and JSX (jsx-a11y)
- Made a few minor adjustments to the eslint-config-payscale index

### CODE REVIEW

**Review by alexdangelo — APPROVED** [review]

File: strict-addons/vanilla-js/index.js [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+    // http://eslint.org/docs/rules/class-methods-use-this
+    'class-methods-use-this': ['error', {
+      exceptMethods: [],
+    }],
+
+    // require return statements to either always or never specify values
+    'consistent-return': 'error',
+
+    // must use curly braces (no if (foo) bar();)
+    curly: ['error', 'all'],
```
Comment (ncua) [inline strict-addons/vanilla-js/index.js]: we're inconsistent with double quote, single quote or no quotes. let's choose one and go with it.

**Review by ncua — COMMENTED** [review]
Things look good other than keeping consistent with double, single or no quotes.

File: strict-addons/vanilla-js/index.js (hunk 1 above)
Comment (PayscaleNateW) [reply strict-addons/vanilla-js/index.js]: ah man. who lints the linter?

Comment (ncua) [reply strict-addons/vanilla-js/index.js]: haha right, i'm surprised lint doesn't just lint itself.

**Pull request merged**